src/System.Private.ServiceModel/tools/IISHostedWcfService/App_code/ComplexCompositeType.cs

```diff
@@ -0,0 +1,275 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Runtime.Serialization;
+using System.Text;
+
+namespace WcfService
+{
+    [DataContract(Namespace = "http://www.contoso.com/wcfnamespace")]
+    public class ComplexCompositeType : IEquatable<ComplexCompositeType>
```

Comment:
All *.cs files are shared by IIS and SelfHost. The SelfHost will include AppCode folder. 

For service interface types and data types between service and client, it's duplicated in the current test infrastructure and we haven't discussed if we should redesign. I think this should be tracked by a separate issue and discuss the design over there.